File: app/src/main/java/com/example/acplite/ui/ActivityProductRegister.java
package com.example.acplite.ui;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.database.Cursor;
import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import com.example.acplite.R;
import com.example.acplite.entidades.Arbol;
import com.example.acplite.entidades.Productos;
import com.example.acplite.sqlite.ConexionSQLiteHelper;
import com.example.acplite.utilidades.UtilidadesArbol;
import com.example.acplite.utilidades.UtilidadesProducto;

import java.io.IOException;

public class ActivityProductRegister extends AppCompatActivity {
    private EditText productID, productName, productPrice;
    private ImageView imageViewObj;

    private static final int PICK_IMAGE_REQUEST = 100;
    private Uri imageFilePath;
    private Bitmap imgToStore;

    private ConexionSQLiteHelper dbh;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_product_register);
        dbh = new ConexionSQLiteHelper(getApplicationContext(), UtilidadesProducto.DB_NAME, null, 1);

        productID = findViewById(R.id.etProductID);
        productName = findViewById(R.id.etProductName);
        productPrice = findViewById(R.id.etProductPrice);
        imageViewObj = findViewById(R.id.productImg);

        imageViewObj.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                chooseImage();
            }
        });
    }

    private void chooseImage() {
        Intent intent = new Intent();
        intent.setType("image/*");

        intent.setAction(Intent.ACTION_GET_CONTENT);
        startActivityForResult(intent, PICK_IMAGE_REQUEST);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable @org.jetbrains.annotations.Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if(requestCode == PICK_IMAGE_REQUEST && resultCode == RESULT_OK && data != null &&
                data.getData() != null){

            imageFilePath = data.getData();
            try {
                imgToStore = MediaStore.Images.Media.getBitmap(getContentResolver(), imageFilePath);
            } catch (IOException e) {
                e.printStackTrace();
            }

            imageViewObj.setImageBitmap(imgToStore);
        }
    }

    public void storeProductImage(View v){
        String id = productID.getText().toString();
        String Name = productName.getText().toString();
        String Price = productPrice.getText().toString();

        if( !Name.isEmpty() && !id.isEmpty() && !Price.isEmpty() && imageViewObj.getDrawable() != null && imgToStore != null){

            Cursor cursor = dbh.validateProduct(id);
            if(cursor.getCount() > 0){
                Toast.makeText(getApplicationContext(), "El Producto ya existe", Toast.LENGTH_SHORT).show();
            } else {
                int campoID = Integer.parseInt(id);
                float campoPrice = Float.parseFloat(Price);
                dbh.storeProducto(new Productos(campoID, Name, campoPrice, imgToStore));

                Toast.makeText(getApplicationContext(), "Producto guardado con éxito", Toast.LENGTH_SHORT).show();

                //LIMPIAMOS CAMPOS
                productID.setText("");
                productName.setText("");
                productPrice.setText("");
                imageViewObj.setImageDrawable(getResources().getDrawable(R.drawable.ic_baseline_camera_alt_24));
            }

        } else {
            Toast.makeText(getApplicationContext(), "Llene los campos antes de guardar", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/example/acplite/ui/ActivityProductRegister.kt
package com.example.acplite.ui

import android.content.Intent
import android.graphics.Bitmap
import android.net.Uri
import android.os.Bundle
import android.provider.MediaStore
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.acplite.R
import com.example.acplite.entidades.Productos
import com.example.acplite.sqlite.ConexionSQLiteHelper
import com.example.acplite.utilidades.UtilidadesProducto
import java.io.IOException

class ActivityProductRegister : AppCompatActivity() {
    private var productID: EditText? = null
    private var productName: EditText? = null
    private var productPrice: EditText? = null
    private var imageViewObj: ImageView? = null
    private var imageFilePath: Uri? = null
    private var imgToStore: Bitmap? = null
    private var dbh: ConexionSQLiteHelper? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_product_register)
        dbh = ConexionSQLiteHelper(applicationContext, UtilidadesProducto.DB_NAME, null, 1)
        productID = findViewById(R.id.etProductID)
        productName = findViewById(R.id.etProductName)
        productPrice = findViewById(R.id.etProductPrice)
        imageViewObj = findViewById(R.id.productImg)
        imageViewObj.setOnClickListener(View.OnClickListener { chooseImage() })
    }

    private fun chooseImage() {
        val intent = Intent()
        intent.type = "image/*"
        intent.action = Intent.ACTION_GET_CONTENT
        startActivityForResult(intent, PICK_IMAGE_REQUEST)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == PICK_IMAGE_REQUEST && resultCode == RESULT_OK && data != null && data.data != null) {
            imageFilePath = data.data
            try {
                imgToStore = MediaStore.Images.Media.getBitmap(contentResolver, imageFilePath)
            } catch (e: IOException) {
                e.printStackTrace()
            }
            imageViewObj!!.setImageBitmap(imgToStore)
        }
    }

    fun storeProductImage(v: View?) {
        val id = productID!!.text.toString()
        val Name = productName!!.text.toString()
        val Price = productPrice!!.text.toString()
        if (!Name.isEmpty() && !id.isEmpty() && !Price.isEmpty() && imageViewObj!!.drawable != null && imgToStore != null) {
            val cursor = dbh!!.validateProduct(id)
            if (cursor!!.count > 0) {
                Toast.makeText(applicationContext, "El Producto ya existe", Toast.LENGTH_SHORT)
                    .show()
            } else {
                val campoID = id.toInt()
                val campoPrice = Price.toFloat()
                dbh!!.storeProducto(Productos(campoID, Name, campoPrice, imgToStore!!))
                Toast.makeText(
                    applicationContext,
                    "Producto guardado con éxito",
                    Toast.LENGTH_SHORT
                ).show()

                //LIMPIAMOS CAMPOS
                productID!!.setText("")
                productName!!.setText("")
                productPrice!!.setText("")
                imageViewObj!!.setImageDrawable(resources.getDrawable(R.drawable.ic_baseline_camera_alt_24))
            }
        } else {
            Toast.makeText(
                applicationContext,
                "Llene los campos antes de guardar",
                Toast.LENGTH_SHORT
            ).show()
        }
    }

    companion object {
        private const val PICK_IMAGE_REQUEST = 100
    }
}